src/main/webapp/dev/js/main/adminHome.es6

```diff
@@ -1,14 +1,156 @@
+        this.message = message;
+        this.originalString = originalString;
+    }
+    toString() {
+        return this.originalString;
+    }
+    /* eslint-disable class-methods-use-this */
+    isError() {
+        return true;
+    }
+    /* eslint-enable class-methods-use-this */
+    getErrorMessage() {
+        return this.message;
+    }
+}
+
+/**
+ * Represents an instructor.
+ * Contains the shortname, name, email and institution fields.
+ * Static "constructor" functions are used since constructors cannot be overloaded in ES6.
+ */
+class Instructor {
+    /**
+     * Takes in several instructor attributes and constructs an instructor.
+     */
+    static create(shortName, name, email, institution) {
+        const instructor = new Instructor();
+        instructor.shortName = shortName;
+        instructor.name = name;
+        instructor.email = email;
+        instructor.institution = institution;
+        return instructor;
+    }
+    /**
+     * Takes in a string in either of the following formats:
+     *  NAME | EMAIL | INSTITUTION
+     *  NAME\tEMAIL\tINSTITUTION
+     * If the string is in the correct format, parses it and constructs an Instructor.
+     * If the format is wrong, constructs and returns a InstructorError instead.
+     */
+    static createFromString(str) {
+        const regexStringForPipeSeparator = '(?: *\\| *)';
+        const regexStringForTabSeparator = '\\t+';
+        const regexStringForSeparator = `(?:${regexStringForTabSeparator}|${regexStringForPipeSeparator})`;
+        const regexStringForFields = '([^|\\t]+?)';
+
+        const instructorMatchRegex = new RegExp(`^${[
+            regexStringForFields, // name
+            regexStringForSeparator,
+            regexStringForFields, // email
+            regexStringForSeparator,
+            regexStringForFields, // institution
+        ].join('')}$`);
+
+        const instructorData = str.match(instructorMatchRegex);
+        if (instructorData === null) {
+            return new InstructorError(Const.StatusMessages.INSTRUCTOR_DETAILS_LENGTH_INVALID, str);
+        }
+
+        const shortName = instructorData[1];
+        const name = instructorData[1];
+        const email = instructorData[2];
+        const institution = instructorData[3];
+        return Instructor.create(shortName, name, email, institution);
+    }
+    toString() {
+        return `${this.name} | ${this.email} | ${this.institution}`;
+    }
+    getParamString() {
+        return $.param({
+            instructorshortname: this.shortName,
+            instructorname: this.name,
+            instructoremail: this.email,
+            instructorinstitution: this.institution,
+        });
+    }
+    /* eslint-disable class-methods-use-this */
+    isError() {
+        return false;
+    }
+    /* eslint-enable class-methods-use-this */
+    static allFromString(multipleInstructorsString) {
+        return multipleInstructorsString
+            .split('\n')
```

Comment:
Can do a simple `.filter` here. In addition to empty lines, it should filter out lines that contain nothing but whitespace too.